.github/workflows/codeql-analysis.yml

```diff
@@ -11,8 +11,15 @@ concurrency:
   group: codeql-analysis-${{ github.event_name }}-${{ github.ref }}
   cancel-in-progress: true
 
+permissions:
+  contents: read
```

Comment:
If we're going to care about the permissions at all, we should probably change the "Workflow permissions" setting to "Read repository contents permission" so this is the default instead of setting it randomly in a handful of workflows.